FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,35 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case let .success((data, response)):
+				if response.statusCode == 200, let json = try? JSONDecoder().decode(Root.self, from: data) {
+					completion(.success(FeedImageMapper.map(json.items)))
+				} else {
+					completion(.failure(Error.invalidData))
+				}
```

Comment:
Both checking the response's `statusCode` and decoding `json` are part of mapping logic, so they can be moved to the already implemented `FeedImageMapper.map` function. By changing `map`'s function type to `(Data, HTTPURLResponse) -> FeedLoader.Result` you can improve readability. 
```suggestion
				completion(FeedImageMapper.map(data, from: response))
```